### PR TITLE
[chore] make sure all contrib tests run

### LINKED ISSUE #9237: Contrib tests are broken
**Describe the bug**
https://github.com/open-telemetry/opentelemetry-collector/pull/9091 contrib tests should have caught the errors that happened in open-telemetry/opentelemetry-collector-contrib/pull/30167 but it did not. We need to investigate why, and check that contrib tests are actually testing.

This is important to avoid delays in the release like it is happening for v0.92.0

### PR DESCRIPTION
**Pull request by atoulme**
Fixes #9237

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -280,7 +280,7 @@ check-contrib:
 	@$(MAKE) -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector/semconv=$(CURDIR)/semconv"
 	@$(MAKE) -C $(CONTRIB_PATH) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector/service=$(CURDIR)/service"
 	@$(MAKE) -C $(CONTRIB_PATH) -j2 gotidy
-	@$(MAKE) -C $(CONTRIB_PATH) test
+	@$(MAKE) -C $(CONTRIB_PATH) gotest
```
Comment (codeboten) [inline Makefile:283]: This makes me think we should change the `test` target in the contrib repo perhaps....

Comment (atoulme) [reply Makefile:283]: I don't know about that, `make test` is defined in `Makefile.Common` so each component can run tests, and `make gotest` is only defined in root. Other tasks follow the same pattern, `gotidy` vs `tidy` for example.

Comment (codeboten) [reply Makefile:283]: Yeah that's fair... test is just more intuitive than gotest for anyone starting with the repo

**Comment by atoulme** [comment]
```
--- FAIL: TestLoadConfig (0.01s)
    config_test.go:40: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/config_test.go:40
        	Error:      	Received unexpected error:
        	            	cannot unmarshal the configuration: 1 error(s) decoding:
        	            	
        	            	* error decoding 'receivers': error reading configuration for "hostmetrics/customname": 1 error(s) decoding:
        	            	
        	            	* '' has invalid keys: scrapers
        	Test:       	TestLoadConfig
--- FAIL: TestLoadInvalidConfig_InvalidScraperKey (0.00s)
    config_test.go:143: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/config_test.go:143
        	Error:      	"cannot unmarshal the configuration: 1 error(s) decoding:\n\n* error decoding 'receivers': error reading configuration for \"hostmetrics\": 1 error(s) decoding:\n\n* '' has invalid keys: scrapers" does not contain "error reading configuration for \"hostmetrics\": invalid scraper key: invalidscraperkey"
        	Test:       	TestLoadInvalidConfig_InvalidScraperKey
--- FAIL: TestLoadConfigRootPath (0.00s)
    hostmetrics_linux_test.go:42: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/hostmetrics_linux_test.go:42
        	Error:      	Received unexpected error:
        	            	cannot unmarshal the configuration: 1 error(s) decoding:
        	            	
        	            	* error decoding 'receivers': error reading configuration for "hostmetrics": 1 error(s) decoding:
        	            	
        	            	* '' has invalid keys: scrapers
        	Test:       	TestLoadConfigRootPath
--- FAIL: TestLoadInvalidConfig_RootPathNotExist (0.01s)
    hostmetrics_linux_test.go:67: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/hostmetrics_linux_test.go:67
        	Error:      	Error "cannot unmarshal the configuration: 1 error(s) decoding:\n\n* error decoding 'receivers': error reading configuration for \"hostmetrics\": 1 error(s) decoding:\n\n* '' has invalid keys: scrapers" does not contain "invalid root_path:"
        	Test:       	TestLoadInvalidConfig_RootPathNotExist
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver	2.470s
```

**Review by bogdandrutu — APPROVED** [review]

**Comment by atoulme** [comment]
This change https://github.com/open-telemetry/opentelemetry-collector/pull/9154 broke the tests.

**Review by mx-psi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Fixed with the latest update in contrib

**Pull request merged**